packages/sitecore-jss-nextjs/src/components/NextImage.test.tsx

```diff
@@ -0,0 +1,318 @@
+/* eslint-disable no-unused-expressions */
+import chai, { use } from 'chai';
+import chaiString from 'chai-string';
+import { mount } from 'enzyme';
+import React from 'react';
+import { NextImage, loader } from './NextImage';
+import { ImageField } from '@sitecore-jss/sitecore-jss-react';
+import { ImageLoader, ImageLoaderProps } from 'next/image';
+import { spy, match } from 'sinon';
+import sinonChai from 'sinon-chai';
+import { SinonSpy } from 'sinon';
+
+use(sinonChai);
+const expect = chai.use(chaiString).expect;
+
+type MockLoaderType = ImageLoader extends SinonSpy ? SinonSpy : SinonSpy<any, any>;
+const customLoader = ({ src }) => new URL(`https://cm.jss.localhost${src}`).href;
+const mockLoader = (spy(customLoader) as unknown) as MockLoaderType;
+
+describe('<NextImage />', () => {
+  afterEach(() => {
+    mockLoader.resetHistory();
+  });
```

Comment:
This `afterEach` is only effective for tests within the `describe` block. I'd recommend we move the `mockLoader` (lines 16-18) inside this describe block, and then also move inside of here every test that uses it (including the wrapping `describe` blocks). Which would be everything other than the 'Next loader function' ones. That gives us 2 outer blocks: 1) '<NextImage />' - tests the render / mount, and 2) 'Next loader function' - tests the loader function

Reply:
Ohh but I tested it. afterEach was running for all the describes nested inside.
BUt what you suggest sounds better.